core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftScanner.java

```diff
@@ -358,7 +380,22 @@ else if (log.isTraceEnabled())
           }
 
           TraceUtil.setException(child2, e, false);
-          sleepMillis = pause(sleepMillis, maxSleepTime);
+          sleepMillis = pause(sleepMillis, maxSleepTime, scanState.runOnScanServer);
+        } catch (ScanServerBusyException e) {
+          error = "Scan failed, scan server was busy " + loc;
```

Comment:
While testing I encountered an exception on the scan server.  The error message on the client side indicated there was an error on the tserver.  That is because this loc is the tserver.  The code that generated the error message I saw is further up, it was from the following line.

```
throw new AccumuloServerException(loc.tablet_location, tae);
```

All of the catch blocks in this try that use the loc may have have the same problem.  We may be able to use `scanState.prevLoc` instead of loc as this is always set to the actual server before making the rpc call.

Reply:
fix applied in ec7cbad